File: front-end/src/app/models/venda/venda.update.dto.ts
import { ItemVenda } from '../itemvenda/itemVenda.model';

export class VendaUpdateDTO {
  idvenda: string;
  data: Date;
  total: number;
  idcliente: string;
  itensvenda: ItemVenda[];

  constructor(
    idvenda: string,
    data: Date,
    idcliente: string,
    total: number,
    itensvenda: ItemVenda[],
  ) {
    this.idvenda = idvenda;
    this.data = data;
    this.idcliente = idcliente;
    this.total = total;
    this.itensvenda = itensvenda;
  }
}

File: back-end/src/main/java/com/example/Intesis/Dto/VendaUpdateDto.java
package com.example.Intesis.Dto;

import com.example.Intesis.Entity.Item_Venda.Item_Venda;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;
import java.util.UUID;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class VendaUpdateDto {
    private UUID idvenda;

    private Date data;

    private Float total;

    private UUID idcliente;

    private List<Item_Venda> itensvenda;
}
